src/pages/api/auth/[...nextauth].ts

```diff
@@ -26,12 +66,35 @@ export const nextAuthOptions: NextAuthOptions = {
                 token.lastName = profile.lastName;
                 token.email = profile.email;
             }
+            if (user) {
+                token.userId = user.id;
+            }
             return token;
         },
         session: async ({ session, token }) => {
             session.firstName = token.firstName;
             session.lastName = token.lastName;
             session.email = token.email;
+            session.userId = token.userId;
+            const user = await getUser(token.userId);
+            if (user == null) {
+                const newUser = {
+                    user_id: token.userId,
+                    first_name: token.firstName,
+                    last_name: token.lastName,
+                    email: token.email,
+                    preferences: [],
```

Comment:
Kinda goes along with the interface discussion, but could we call this `followed_topics` or something? I feel like we might have more preferences that the user can set in the future, and it could get confusing if the `preferences` field is just the topics they follow.